src/chapters/angular-lsn1/inside-app-folder.rst

```diff
@@ -0,0 +1,289 @@
+         imports: [ BrowserModule ],
+         providers: [],
+         bootstrap: [AppComponent]
+      })
+      export class AppModule { }
+
+Just like before, there is a lot going on within very few lines.
+
+#. Lines 1, 2, and 8 import and assign the core modules that make Angular
+   work. This is part of the automatic process, so do not play with these
+   (yet).
+#. Line 4 imports the class ``AppComponent`` from the local file
+   ``app.component.ts``.
+#. Line 4 also pulls in references to any other files linked to
+   ``app.component.ts``.
+#. Line 7 declares the imported local files as necessary for the project.
+#. Line 12 exports the ``AppModule`` class and makes it available to other
+   files.
+
+``app.module.ts`` does the main work of pulling in the core libraries and local
+files. As new parts are added to a project, the import statements, ``imports``
+array, and ``declarations`` array update automatically. We do not have to worry
+about the details for adding this critical code ourselves.
+
+Change The Content
+-------------------
+
+Enough detail. Let's explore some more.
+
+If you did not complete all of the :ref:`Try It <try-it-Angular-intro>`
+tasks on the previous page, attempt them now. After that...
+
+Try It!
+^^^^^^^^
+
+#. Run ``ng serve`` in the terminal to launch your webpage again.
+#. In ``app.component.ts``, declare and assign two variables in the
+   ``AppComponent`` class---``name`` and ``itemList``.
+
+   a. ``name`` holds your name.
+   b. ``itemList`` is an array holding at least 4 items.
+
+   .. sourcecode:: TypeScript
+      :linenos:
+
+      export class AppComponent {
+         name: string = 'Barbara Liskov';
+         itemList: [string, string, string, string] = ['item1', 'item2', 'item3', 'item4'];
+      }
+
+   .. admonition:: Note
+
+      Instead of using the strong TypeScript variable declarations in step 2, we
+      could substitute a pattern more like JavaScript:
+
+      .. sourcecode:: JavaScript
+         :linenos:
+
+         export class AppComponent {
+            name = 'Brendan Eich';
+            itemList = ['item1', 'item2', 'item3', 'item4'];
+         }
+
+#. Replace line 4 in ``app.component.html`` with ``<h1>{{name}}'s First
+   Angular Project</h1>``. Save your work and then check to make sure the
+   webpage shows the new heading.
+#. Modify the ``<li></li>`` elements in ``app.component.html`` to display the
+   elements from ``itemList`` in an unordered list. Be sure to use
+   placeholders like ``{{itemList[0]}}`` between the tags.
+#. Define a ``rectangle`` object in ``AppComponent`` that has keys of
+   ``length``, ``width`` and ``area``. Assign numbers to ``length`` and
+   ``width``, and have ``area`` be a function that calculates and returns the
+   area.
+
+   .. sourcecode:: TypeScript
+      :linenos:
+
+      rectangle = {
+         length: 5,
+         width: 6,
+         area: function() {
+            return this.length*this.width;
```

Comment:
Please change to `return this.length * this.width;`, the extra space makes it more readable.

Reply:
Got it.  Done.